modules/reranker-cohere/clients/ranker_test.go

```diff
@@ -160,16 +161,22 @@ func TestRank(t *testing.T) {
 }
 
 type testRankHandler struct {
+	lock           sync.RWMutex
 	t              *testing.T
 	response       RankResponse
 	batchedResults [][]Result
 	errorMessage   string
 }
 
 func (f *testRankHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if f.errorMessage != "" {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	response, batchedResults, errorMessage := f.response, f.batchedResults, f.errorMessage
```

Comment:
I suppose lock have to assure only one ServeHTTP can be executed at a time.
Does var assignment have some purpose here?

Reply:
I have done more tests with `-race` and it looks like this is not needed, locking solves the Data Race problem.
I will push a fix for this